File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int lh=0,rh=height.size()-1,maxWater=0,area=0;
        while(lh<rh){
            int ht = min(height[lh],height[rh]);
            int length = (rh-lh);
            area = ht*length;
            maxWater=max(maxWater,area);
           height[lh]<height[rh] ? lh++ :rh--;
        }
        return maxWater;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left=0
        right=len(height)-1
        maxArea=0
        while left<right:
            h=min(height[left],height[right])
            l=right-left
            area=h*l
            maxArea=max(maxArea,area)
            if height[left]>height[right]:
                right-=1
            else:
                left+=1
        return maxArea
        